### ISSUE TITLE
Zathura preview problem

### ISSUE THREAD

**Issue by Vintodrimmer** [lervag/vimtex#901]
Good day,

I'm running `neovim` with `vimtex` through `vim-plug`. Complete list of plugins is as follows: 
```
 17 Plug 'lervag/vimtex'
 18 Plug 'w0rp/ale'
 19 Plug 'itchyny/lightline.vim'
```

There might be some problem with the `zathura` viewing method. Since sometimes `zathura` would close and `firefox` would open, or I'd get multiple instances, I decided to explicitly set `let g:vimtex_view_method = 'zathura'`

However, then I simply got 

> vimtex warning: viewer can not wind Zathura window ID!

If I run the instance of `zathura` in the background, it issues no warning, but also does not show anything.

If I set `let g:vimtex_view_general_viewer = 'zathura'` instead, everything works as expected.

Please, look into it if you have time. It does work now, but I'd like to get the correct implementation, since `zathura` has its own mode.

**Comment by lervag** [comment]
1. You should use `let g:vimtex_view_method = 'zathura'`.
2. You have not really specified a reproducible problem. Please read the [issue template](https://github.com/lervag/vimtex/blob/master/ISSUE_TEMPLATE.md) and try again.

**Comment by lervag** [comment]
Just to be clear: You did not 1) provide a minimal vimrc file, 2) provide a detailed explanation of what you problem is, and in particular, 3) you did not provide a list of steps required to produce your issue along with a description of i) observed behaviour and ii) expected behaviour.

I can not help you if you do not provide enough information for me to understand the problem. If it is a bug, then I need enough information to reproduce it. I think this is not a bug, but instead a problem with your configuration (Vim and/or system), and it is therefore relevant to hear more about your system.

Let me know if you need help to create and run a minimal example. I think this should be fairly well described in the issue template, though.

**Comment by Vintodrimmer** [comment]
Sorry, I tried to be as specific as possible. Let me try again.

### The problem:
If I set `let g:vimtex_view_method = 'zathura'` in my neovim config, `zathura` stops showing previews and gives an error message about window ID.

### How to reproduce:

1. write `let g:vimtex_view_method = 'zathura'` to your `nvim.init`.
2. Open the `.tex` file
3. Press `lv` in the normal mode to open the viewer. (assuming you have already compiled the .pdf}

#### Observed behaviour:
> vimtex warning: viewer can not find Zathura window ID!
![vimtex warning](https://raw.githubusercontent.com/Vintodrimmer/docs/master/2017-07-11-08%3A26%3A26-screenshot.png)

#### Expected behaviour:
a new instance of `zathura` with the file preview.

### Alternatively:

1. write `let g:vimtex_view_method = 'zathura'` to your `nvim.init`.
2. Open the `.tex` file
3. Open instance of `zathura` independently
3. Press `lv` in the normal mode to open the viewer. (assuming you have already compiled the .pdf}

#### Observed behaviour:
Nothing happens. At all.
#### Expected behaviour:
Preview in the opened instance, maybe? Not sure what I expected.

### Minimal config with which it happens is as follows:

```
set nocompatible

" Load vim-plug with vimtex
call plug#begin('~/.vim/plugged')
Plug 'lervag/vimtex'
call plug#end()


let g:vimtex_view_method = 'zathura'
```

### Temporary solution: 

Use `let g:vimtex_view_general_viewer = 'zathura'`  instead.

Hope this makes things clearer.

**Comment by lervag** [comment]
I've pushed some test files that correspond to your description. I run `nvim -u minivimrc` and follow your steps. Everything works as expected. If I do `vim -u minivimrc` it also works perfectly.

Note:

* In my tests, I do not use `vim-plug`, I use a _real_ minimal vimrc file.
* I've tested with both neovim 0.2.0 and Vim 8.0.624.

----

So: As I tried to explain before, the behaviour you are observing may be due to some configuration of your system. Please provide more details, for instance:

- Which version of Vim?
- Which version of Zathura?
- Which OS?
- Everything else that might be relevant.

Also, please try with the minimal example that I just pushed. If you update the vimtex plugin you should find the example under `VIMTEX/test/issues/901`, and in this directory you can start the example with `vim -u minivimrc` (it will automatically load the `minimal.tex` file). Then you can try `<leader>lv` to open the pdf in Zathura.

**Comment by Vintodrimmer** [comment]
I think, I might be doing something very stupid, but when I start it with your configs:

```
cd ~/.vim/plugged/vimtex/test/issues/901
cp minivimrc ~/.config/nvim/init.vim 
vim minimal.tex
```

Nothing happens, when I press `\lv`, it just starts `VISUAL` mode. I think that `vimtex` isn't loaded, since there is no help file for it either.

Neovim version:
```
:version
NVIM v0.2.0
Build type: Release
Compilation: /usr/lib/ccache/bin/cc -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -
mtune=generic -O2 -pipe  -g -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HAS_FLOAT32 -O2 -DNDEBUG -DDISABLE_LOG -Wall -Wextra -pedanti
c -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOUR
CE -I/builddir/neovim-0.2.0/build/config -I/builddir/neovim-0.2.0/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/inc
lude -I/builddir/neovim-0.2.0/build/src/nvim/auto -I/builddir/neovim-0.2.0/build/include
Compiled by void-buildslave@build

Optional features included (+) or not (-): +acl   +iconv    -jemalloc +tui
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

Zathura version:

```
zathura 0.3.7
(plugin) pdf-mupdf (0.3.1) (/usr/lib/zathura/pdf.so)
```

OS is Void Linux.

I'm not sure that there is anything else relevant, to be honest. I have installed `vim` (und uninstalled `nvim`, so I can try it without changing the `minivimrc`) to test it out and the result is the same (vimtex not loading. But now `vim` beeps, when I press `\`).

Version is:

```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jun 27 2017 12:11:55)
Included patches: 1-679
Compiled by void-buildslave@build
Huge version without GUI.  Features included (+) or not (-):
+acl             +comments        +extra_search    +keymap          +mouse_dec       +path_extra      +smartindent     +title           +xfontset
+arabic          +conceal         +farsi           +lambda          -mouse_gpm       -perl            +startuptime     -toolbar         -xim
+autocmd         +cryptv          +file_in_path    +langmap         -mouse_jsbterm   +persistent_undo +statusline      +user_commands   -xpm
-balloon_eval    +cscope          +find_in_path    +libcall         +mouse_netterm   +postscript      -sun_workshop    +vertsplit       +xsmp_interact
-browse          +cursorbind      +float           +linebreak       +mouse_sgr       +printer         +syntax          +virtualedit     +xterm_clipboard
++builtin_terms  +cursorshape     +folding         +lispindent      -mouse_sysmouse  +profile         +tag_binary      +visual          -xterm_save
+byte_offset     +dialog_con      -footer          +listcmds        +mouse_urxvt     -python          +tag_old_static  +visualextra
+channel         +diff            +fork()          +localmap        +mouse_xterm     -python3         -tag_any_white   +viminfo
+cindent         +digraphs        +gettext         -lua             +multi_byte      +quickfix        -tcl             +vreplace
+clientserver    -dnd             -hangul_input    +menu            +multi_lang      +reltime         +termguicolors   +wildignore
+clipboard       -ebcdic          +iconv           +mksession       -mzscheme        +rightleft       +terminfo        +wildmenu
+cmdline_compl   +emacs_tags      +insert_expand   +modify_fname    -netbeans_intg   -ruby            +termresponse    +windows
+cmdline_hist    +eval            +job             +mouse           +num64           +scrollbind      +textobjects     +writebackup
+cmdline_info    +ex_extra        +jumplist        -mouseshape      +packages        +signs           +timers          +X11
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: cc -c -I. -Iproto -DHAVE_CONFIG_H     -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -mtun
e=generic -O2 -pipe   -g -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: cc   -specs=/void-packages/common/environment/configure/gccspecs/hardened-ld -Wl,-z,relro -Wl,-z,now     -L/usr/local/lib -Wl,--as-needed -o vim
  -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE  -lm -lelf -lnsl  -lncursesw -lacl -lattr -ldl
```

Sorry for bugging you, but I would honestly want to figure out the problem.

**Comment by lervag** [comment]
Ok, some more pointers:

* You can have Vim and neovim installed at the same time, no problem. No reason to uninstall anything. You start neovim with `nvim` and Vim with `vim`.

* Instead of copying the `minivimrc` to your `init.vim` or `vimrc` file, you can do `vim -u minivimrc` or `nvim -u minivimrc`. Thus, instead of:

    ```sh
    cd ~/.vim/plugged/vimtex/test/issues/901
    cp minivimrc ~/.config/nvim/init.vim 
    vim minimal.tex
    ```

    you can do

    ```sh
    cd ~/.vim/plugged/vimtex/test/issues/901
    nvim -u minivimrc
    # or
    vim -u minivimrc
    ```

* When you have started the minimal example, you can do `map \lv` to see the what is currently mapped to `\lv`. You can also check if Vimtex is loaded through the commands, e.g. `:VimtexInfo`.

**Comment by Vintodrimmer** [comment]
Thanks for the pointers! it's much more convenient, without the config copying.

When I do 

```
cd ~/.vim/plugged/vimtex/test/issues/901
nvim -u minivimrc
```
 
then: `map \lv ` returns `No mapping found` and `:VimtexInfo` returns `E492: Not an editor command: VimtexInfo`

Same story with `vim`.

So it does seem that `vimtex` is not loaded with the minimum config. May that have something to do with the initial problem?

**Comment by lervag** [comment]
Ah, no: You need to update the `minivimrc` file: Change `bundle` to `plugged` and try again. Sorry for not noticing before.

**Comment by Vintodrimmer** [comment]
Thank you.

But I'm afraid the problem persists in both `vim` and `nvim` even with minimal config provided by you.

![the problem](https://raw.githubusercontent.com/Vintodrimmer/docs/master/2017-07-11-22%3A55%3A49-screenshot.png)

`:VimtexInfo`
```

vimtex project: minimal
  base: minimal.tex
  root: /home/eichhorn/.vim/plugged/vimtex/test/issues/901
  tex: /home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.tex
  out: /home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.pdf
  log: 
  aux: 
  fls: 
  document class: minimal
  compiler: latexmk
    backend: nvim
    output: /tmp/nvimRElWZv/1
    configuration: 
      continuous: 1
      callback: 1
      latexmk options: 
        -verbose
        -pdf
        -file-line-error
        -synctex=1
        -interaction=nonstopmode
  viewer: Zathura
    xwin id: 0
    process: 
      pid: -
      cmd: zathura -x "nvr --servername /tmp/nvimRElWZv/0 --remote +\%{line}
 \%{input}" --synctex-forward 1:1:'/home/eichhorn/.vim/plugged/vimtex/test/i
ssues/901/minimal.tex'  '/home/eichhorn/.vim/plugged/vimtex/test/issues/901/
minimal.pdf' >/dev/null 2>&1 &
  qf: LaTeX logfile
    config: 
      fix_paths: 1
      packages: 
        default: 1
      default: 1

```

**Comment by lervag** [comment]
Thanks, now we are getting somewhere. Do you have `xdotool` installed?

**Comment by Vintodrimmer** [comment]
Yes. If I remember correctly, `vimtex` said I need it for the forward search, the first time I tried to enable `let g:vimtex_view_method = 'zathura'`, after which I have installed it.

**Comment by lervag** [comment]
Ok. Could you try to run the following command from the terminal:

```
zathura -x "nvr --servername /tmp/nvimRElWZv/0 --remote +\%{line} \%{input}" --synctex-forward 1:1:'/home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.tex' '/home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.pdf' >/dev/null 2>&1 &
```

Note: Change the `nvimREIWZv/0` to the corresponding output of `:echo v:servername` from the current `neovim` instance.

**Comment by Vintodrimmer** [comment]
```
 zathura -x "nvr --servername /tmp/nvimpxKdP3/0 --remote +\%{line} \%{input}" --synctex-forward 1:1:'/home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.tex' '/home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.pdf' >/dev/null 2>&1 &
[1] 29133
[1]  + exit 255   zathura -x "nvr --servername /tmp/nvimpxKdP3/0 --remote +\%{line} \%{input}"   
```

**Comment by lervag** [comment]
Ok, remove the last `&` and the `>/dev/null 2>&1` part and repeat.

**Comment by Vintodrimmer** [comment]
```
zathura -x "nvr --servername /tmp/nvimpxKdP3/0 --remote +\%{line} \%{input}" --synctex-forward 1:1:'/home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.tex' '/home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.pdf'
error: Error parsing command line arguments: Unknown option -x
```

**Comment by lervag** [comment]
Ah, there it is. It seems you have a version of Zathura that does not support synctex. If you compiled it yourself, you need to have `libsynctex` installed in order for Zathura to have the feature. If not, then I would alert the package maintainer and make him/her update the package to include the synctex support.

In the meantime, you can solve this by installing Zathura from the source. Make sure you have the synctex dependencies installed first, though.

**Comment by Vintodrimmer** [comment]
Thank you very much for the assistance! It was a pleasure.

I will compile it from the source then and ask maintainers to fix the package if they are willing to.

**Comment by lervag** [comment]
No problem! Let me know if this does not fix your issue (although I am quite certain that it will). Note that you can see if the synctex capabilities are present with `zathura -h`, which outputs for me:

```
> zathura -h
Usage:
  zathura [OPTION…]  [file1] [file2] [...]

Help Options:
  -h, --help                           Show help options

Application Options:
  -e, --reparent=xid                   Reparents to window specified by xid (X11)
  -c, --config-dir=path                Path to the config directory
  -d, --data-dir=path                  Path to the data directory
  --cache-dir=path                     Path to the cache directory
  -p, --plugins-dir=path               Path to the directories containing plugins
  --fork                               Fork into the background
  -w, --password=password              Document password
  -P, --page=number                    Page number to go to
  -l, --log-level=level                Log level (debug, info, warning, error)
  -v, --version                        Print version information
  -x, --synctex-editor-command=cmd     Synctex editor (forwarded to the synctex command)
  --synctex-forward=position           Move to given synctex position
  --synctex-pid=pid                    Highlight given position in the given process
  --mode=mode                          Start in a non-default mode

```

I have Zathura version 0.3.7 also.

**Comment by dolefeast** [comment]
Hi I have the same problem that @Vintodrimmer, but when running 
`zathura -x "nvr --servername /tmp/nvimRElWZv/0 --remote +\%{line} \%{input}" --synctex-forward 1:1:'/home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.tex' '/home/eichhorn/.vim/plugged/vimtex/test/issues/901/minimal.pdf'`
I get 
`error: could not open plugin directory: /usr/lib/zathura                        
error: Could not determine file type. `

I used your minimal vimrc and got the same error, `Viewer cannot find Zathura window ID!`. I'm using vim version 8.2 with zathura version 0.4.7 in Arch Linux.
By the way, when running `zathura --version` I get as ouptut the same error, 
`error: could not open plugin directory: /usr/lib/zathura`
so that should be the problem?

Please let me know if I should upload something else, open a new issue, or whatever else, since I don't usually use git and my understanding of linux is kind of limited.

**Comment by lervag** [comment]
No problem. I came to answer just that: you should check which zathura backend you have installed, see e.g. [this Arch forum thread](https://bbs.archlinux.org/viewtopic.php?id=220572).
